### PR TITLE
feat(inputs.vsphere): Adding vSAN extension to vSphere plugin

### PR DESCRIPTION
**Pull request by gangadharaswamy**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves [#5966](https://github.com/influxdata/telegraf/issues/5966)

This resumes the old PR [#5970](https://github.com/influxdata/telegraf/pull/5970).

### CODE REVIEW

**Comment by gangadharaswamy** [comment]
@srebhan Could you please review this PR?

File: plugins/inputs/vsphere/vsan.go [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+		}
+		perfSpecs = append(perfSpecs, perfSpec)
+
+		perfRequest := vsantypes.VsanPerfQueryPerf{
+			This:       perfManagerRef,
+			QuerySpecs: perfSpecs,
+			Cluster:    &clusterRef.ref,
+		}
+		resp, err := vsanmethods.VsanPerfQueryPerf(ctx, vsanClient, &perfRequest)
+
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:251]: Please group code that logically belongs together and avoid empty lines between those elements. Like in this example, do not put a newline between assigning `err` and checking it.

File: plugins/inputs/vsphere/vsan.go [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+			}
+			// 2. Iterate on each measurement
+			for _, counter := range em.Value {
+				metricLabel := internal.SnakeCase(counter.MetricId.Label)
+				// 3. Iterate on each data point.
+				for i, values := range strings.Split(counter.Values, ",") {
+					ts := timeStamps[i]
+					if ts == (time.Time{}) {
+						continue
+					}
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:299]: This is very fragile. If you get your timestamps in "unexpected" format for at least _one_ timestamp, you might end-up with a `timeStamps` array _shorter_ than the `values` array and thus panic in line 296! Please be safe here and rather error out than rely on properties that are not _guaranteed_ by code.

File: plugins/inputs/vsphere/vsan.go [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+		&vsantypes.VsanQuerySpaceUsage{
+			This:    spaceManagerRef,
+			Cluster: clusterRef.ref,
+		})
+	if err != nil {
+		return err
+	}
+	fields := make(map[string]interface{})
+	fields["free_capacity_byte"] = resp.Returnval.FreeCapacityB
+	fields["total_capacity_byte"] = resp.Returnval.TotalCapacityB
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:342]:
```suggestion
	fields := map[string]interface{}{
		"free_capacity_byte": resp.Returnval.FreeCapacityB,
		"total_capacity_byte": resp.Returnval.TotalCapacityB,
	}
```
Same for other static inits.

**Review by srebhan — COMMENTED** [review]
Thanks for this nice PR @gangadharaswamy! I have some comments in the code, the most severe ones are probably not checking all returned errors in the unit-test and some potential locations for panics. Please check the comments. Looking forward to your update!

File: plugins/inputs/vsphere/README.md [hunk 4]
```diff
@@ -8,9 +8,12 @@ vCenter servers.
...
 * VMs
 * Datastores
+* vSAN
 
 ## Supported versions of vSphere
 
+This plugin supports vSphere version 5.5 through 7.0.
+
 This plugin supports vSphere version 6.5, 6.7 and 7.0. It may work with versions
 5.1, 5.5 and 6.0, but neither are officially supported.
```
Comment (srebhan) [inline plugins/inputs/vsphere/README.md:18]: Those two parts contradict each other. Which one is correct?

File: plugins/inputs/vsphere/README.md [hunk 5]
```diff
@@ -532,6 +542,220 @@ For a detailed list of commonly available metrics, please refer to
 * virtualDisk stats for VM
   * disk (name of virtual disk)
 
+# VMware vSphere Input Plugin - vSAN extension
+
+vSAN resource is a special type of resource that can be collected by the plugin.
+The configuration of vSAN resource is slightly different from hosts, vms and other resources.
+
+## Prerequisites
```
Comment (srebhan) [inline plugins/inputs/vsphere/README.md:550]: Can you please integrate this more with the existing text? This feels a bit alien here especially when using a first-level heading which reads like this is docu for another plugin.

File: plugins/inputs/vsphere/vsan.go [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/methods"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+	vsanmethods "github.com/vmware/govmomi/vsan/methods"
+	vsantypes "github.com/vmware/govmomi/vsan/types"
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:19]: Please order into _builtin_,  _3rd party_ and _Telegraf internal_ libraries each block in alphabetical order...
```suggestion

	"github.com/vmware/govmomi/object"
	"github.com/vmware/govmomi/vim25"
	"github.com/vmware/govmomi/vim25/methods"
	"github.com/vmware/govmomi/vim25/soap"
	"github.com/vmware/govmomi/vim25/types"
	vsanmethods "github.com/vmware/govmomi/vsan/methods"
	vsantypes "github.com/vmware/govmomi/vsan/types"

	"github.com/influxdata/telegraf"
	"github.com/influxdata/telegraf/internal"
```

File: plugins/inputs/vsphere/vsan.go [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+
+// collectVsan is the entry point for vsan metrics collection
+func (e *Endpoint) collectVsan(ctx context.Context, resourceType string, acc telegraf.Accumulator) error {
+	lower, err := versionLowerThan(e.apiVersion, "5.5")
+	if err != nil {
+		return fmt.Errorf("Failed to get the vCenter version for vSAN: %v", err)
+	}
+	if lower {
+		return fmt.Errorf("A minimum API version of 5.5 is required for vSAN. Found: %s. Skipping VCenter: %s", e.apiVersion, e.URL.Host)
+	}
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:47]: Error text must start lowercase!

File: plugins/inputs/vsphere/vsan.go [hunk 8]
```diff
@@ -0,0 +1,534 @@
...
+	if err != nil {
+		return fmt.Errorf("fail to get client when collect vsan: %v", err)
+	}
+	// Create vSAN client
+	vimClient := client.Client.Client
+	vsanClient := vimClient.NewServiceClient(vsanPath, vsanNamespace)
+	// vSAN Metrics to collect
+	metrics := e.getVsanMetadata(ctx, vsanClient, res)
+	// Iterate over all clusters, run a goroutine for each cluster
+	te := NewThrottledExecutor(e.Parent.CollectConcurrency)
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:61]: Do we need to recreate those elements like clients etc. in every gather cycle? Why not simply create it at `Init()` and use it here?

File: plugins/inputs/vsphere/vsan.go [hunk 9]
```diff
@@ -0,0 +1,534 @@
...
+func (e *Endpoint) vsanEnabled(ctx context.Context, clusterObj *object.ClusterComputeResource) bool {
+	config, err := clusterObj.Configuration(ctx)
+	if err != nil {
+		return false
+	}
+	enabled := config.VsanConfigInfo.Enabled
+	if enabled == nil {
+		return false
+	}
+	return *enabled
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:117]: Maybe
```suggestion
	return enabled != nil && *enabled
```

**Review by srebhan — COMMENTED** [review]
@gangadharaswamy there are still comments from the last review. Please also check the linter errors, you can also see them on your side when running `make lint-branch`!

**Comment by yogeshprasad** [comment]
@srebhan can you please have a look into this as @gangadharaswamy has fixed the review comments?

File: plugins/inputs/vsphere/vsan.go [hunk 10]
```diff
@@ -0,0 +1,532 @@
...
+
+// collectVsan is the entry point for vsan metrics collection
+func (e *Endpoint) collectVsan(ctx context.Context, resourceType string, acc telegraf.Accumulator) error {
+	lower, err := versionLowerThan(e.apiVersion, "5.5")
+	if err != nil {
+		return fmt.Errorf("failed to get the vCenter version for vSAN: %v", err)
+	}
+	if lower {
+		return fmt.Errorf("a minimum API version of 5.5 is required for vSAN. Found: %s. Skipping vCenter: %s", e.apiVersion, e.URL.Host)
+	}
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:48]: Please use github.com/coreos/go-semver/semver for this comparison instead of implementing your own...
```suggestion
	version, err := semver.NewVersion(e.apiVersion)
	if err != nil {
		return fmt.Errorf("failed to get the vCenter version for vSAN: %v", err)
	}
	if version.LessThan(semver.Version{Major: 5, Minor: 5}) 
		return fmt.Errorf("a minimum API version of 5.5 is required for vSAN. Found: %s. Skipping vCenter: %s", e.apiVersion, e.URL.Host)
	}
```

**Review by srebhan — CHANGES_REQUESTED** [review]
@gangadharaswamy and @yogeshprasad there are still some comments open which are not addressed yet. Especially handling the length of timestamp arrays correctly is a must. Please check the past and current comments which are not marked as resolved! They still do apply!

**Comment by srebhan** [comment]
@gangadharaswamy can please run `make docs` and fix the CI/build errors!?

**Comment by powersj** [comment]
Hi @gangadharaswamy,

For the tests, you need to add the new package to the list of our dependencies:

```diff
diff --git a/docs/LICENSE_OF_DEPENDENCIES.md b/docs/LICENSE_OF_DEPENDENCIES.md
index 030430695..2a5d22095 100644
--- a/docs/LICENSE_OF_DEPENDENCIES.md
+++ b/docs/LICENSE_OF_DEPENDENCIES.md
@@ -78,6 +78,7 @@ following works:
 - github.com/awslabs/kinesis-aggregation/go [Apache License 2.0](https://github.com/awslabs/kinesis-aggregation/blob/master/LICENSE.txt)
 - github.com/benbjohnson/clock [MIT License](https://github.com/benbjohnson/clock/blob/master/LICENSE)
 - github.com/beorn7/perks [MIT License](https://github.com/beorn7/perks/blob/master/LICENSE)
+- github.com/blang/semver [MIT License](https://github.com/blang/semver/blob/master/LICENSE)
 - github.com/blues/jsonata-go [MIT License](https://github.com/blues/jsonata-go/blob/main/LICENSE)
 - github.com/bmatcuk/doublestar [MIT License](https://github.com/bmatcuk/doublestar/blob/master/LICENSE)
 - github.com/caio/go-tdigest [MIT License](https://github.com/caio/go-tdigest/blob/master/LICENSE)
```

Additionally it looks like you have a failing lint result, as the `queryPerformance` function is always returning nil:

```
  Error: (*Endpoint).queryPerformance - result 0 (error) is always nil (unparam)
```

Thanks!

**Comment by srebhan** [comment]
Or even better use: github.com/coreos/go-semver...

**Comment by powersj** [comment]
Hi @gangadharaswamy,

Looks like your latest push is failing the tests. Could you look at fixing those up?

Thanks!

**Comment by gangadharaswamy** [comment]
@powersj Thank you! Will look into the issues.

File: plugins/inputs/vsphere/sample.conf [hunk 11]
```diff
@@ -136,6 +136,12 @@
   datacenter_metric_exclude = [ "*" ] ## Datacenters are not collected by default.
   # datacenter_instances = false ## false by default
 
+  ## VSAN
+  vsan_metric_include = [] ## if omitted or empty, all metrics are collected
+  vsan_metric_exclude = [ "*" ] ## vSAN are not collected by default.
+  ## Whether to skip verifying vSAN metrics against the ones from GetSupportedEntityTypes API.
+  vsan_metric_skip_verify = false ## false by default.
```
Comment (Hipska) [inline plugins/inputs/vsphere/sample.conf:143]: Default values should be commented out..
```suggestion
  # vsan_metric_skip_verify = false ## false by default.
```

File: plugins/inputs/vsphere/README.md [hunk 12]
```diff
@@ -659,3 +891,14 @@ vsphere_host_net,clustername=DC0_C0,esxhostname=DC0_C0_H0,host=host.example.com,
 vsphere_host_mem,clustername=DC0_C0,esxhostname=DC0_C0_H0,host=host.example.com,moid=host-30,os=Mac,source=DC0_C0_H0,vcenter=localhost:8989 usage_average=116.21 1535660339000000000
 vsphere_host_net,clustername=DC0_C0,esxhostname=DC0_C0_H0,host=host.example.com,moid=host-30,os=Mac,source=DC0_C0_H0,vcenter=localhost:8989 bytesRx_average=726i,bytesTx_average=643i,usage_average=1504i 1535660339000000000
 ```
+
+## vSAN Sample Output
+
+```shell
```
Comment (Hipska) [inline plugins/inputs/vsphere/README.md:897]: This isn't a shell script, pleas change to `text`.
```suggestion
```text
```

File: plugins/inputs/vsphere/vsphere.go [hunk 13]
```diff
@@ -179,21 +181,24 @@ func init() {
 			DatastoreMetricInclude:    nil,
 			DatastoreMetricExclude:    nil,
 			DatastoreInclude:          []string{"/*/datastore/**"},
+			VSANMetricInclude:         nil,
+			VSANMetricExclude:         nil,
```
Comment (Hipska) [inline plugins/inputs/vsphere/vsphere.go:185]: Shouldn't it be like this according to the README saying this is the default?
```suggestion
			VSANMetricExclude:         []string{"*"},
```

**Review by Hipska — CHANGES_REQUESTED** [review]

File: plugins/inputs/vsphere/README.md [hunk 14]
```diff
@@ -551,9 +561,231 @@ For a detailed list of commonly available metrics, please refer to
...
+    "performance.cmmds-world-cpu",
+    "performance.host-cpu",
+    "performance.host-domowner",
+    "performance.host-memory-slab",
+    "performance.host-memory-heap",
+    "performance.system-mem",
+  ]
+  # by default vsan_metric_skip_verify = false
+  vsan_metric_skip_verify = true
+  # vsan_metric_exclude = ["*"]
```
Comment (Hipska) [inline plugins/inputs/vsphere/README.md:628]: So all the mentioned metrics above will be excluded? (as `*` is the default)

Comment (gangadharaswamy) [reply plugins/inputs/vsphere/README.md:628]: Yes, the user can enable these metrics on the requirement. It depends on the environment.

Comment (Hipska) [reply plugins/inputs/vsphere/README.md:628]: But they will all be excluded because of the *. The example config is wrong IMHO.

**Comment by Hipska** [comment]
@gangadharaswamy Can you please alaborate? Either the documentation is wrong, or the sample config is wrong IMHO.

File: plugins/inputs/vsphere/README.md (hunk 14 above)
Comment (gangadharaswamy) [reply plugins/inputs/vsphere/README.md:628]: > So all the mentioned metrics above will be excluded? (as `*` is the default)

Excludes metrics only when `vsan_metric_exclude = ["*"]` is uncommented. By default `vsan_metric_include` is used to collect the metrics.

Comment (Hipska) [reply plugins/inputs/vsphere/README.md:628]: The example config should show the defaults when commented. The example config currently shows * is the default for metric_exclude, which means all metrics get excluded.

If that is not the case, the sample config should show this:
```toml
# vsan_metric_exclude = []
```

Comment (powersj) [reply plugins/inputs/vsphere/README.md:628]: > The example config currently shows * is the default for metric_exclude, which means all metrics get excluded.

That is correct, we want vsan metrics to be opt-in. This is similar to what you will see for `datacenter_metric_*` and `custom_attribute_*`

Comment (Hipska) [reply plugins/inputs/vsphere/README.md:628]: But as soon as you put some values in the include, the default value of exclude isn't `*` anymore?

Comment (powersj) [reply plugins/inputs/vsphere/README.md:628]: Please point to code you are concerned about.

Comment (Hipska) [reply plugins/inputs/vsphere/README.md:628]: I did 🤷

Comment (powersj) [reply plugins/inputs/vsphere/README.md:628]: And we have concluded that this sample config options is consistent with the other metrics this plugin collects.

Comment (Hipska) [reply plugins/inputs/vsphere/README.md:628]: So the following config will load the summary health?
```toml
  vsan_metric_include = ["summary.health"]
# vsan_metric_exclude = ["*"]
```
What about this one?
```toml
  vsan_metric_include = ["summary.health"]
  vsan_metric_exclude = ["*"]
```
(include summary, but exclude all?)

Comment (gangadharaswamy) [reply plugins/inputs/vsphere/README.md:628]: @Hipska You are right, this should be the sample config.

```
  vsan_metric_include = []
  # by default vsan_metric_skip_verify = false
  vsan_metric_skip_verify = true
  # vsan_metric_exclude = [ ]
  vsan_cluster_include = [ "/*/host/**" ]
```

The config
```
  vsan_metric_include = ["summary.health"]
# vsan_metric_exclude = [ ]
```
OR
```
  vsan_metric_include = ["summary.health"]
  vsan_metric_exclude = [ ]
```
collects `summary.health` metric.

**Comment by gangadharaswamy** [comment]
The `vsan_metric_include = []` or `vsan_metric_include = [<list_of_metrics>]` and `vsan_metric_exclude = [ ]` commented or uncommented collects vSAN metrics.
`vsan_metric_exclude = [ "*" ]` excludes all vSAN metrics.

File: plugins/inputs/vsphere/README.md [hunk 15]
```diff
@@ -551,9 +561,231 @@ For a detailed list of commonly available metrics, please refer to
...
+    "performance.cmmds-world-cpu",
+    "performance.host-cpu",
+    "performance.host-domowner",
+    "performance.host-memory-slab",
+    "performance.host-memory-heap",
+    "performance.system-mem",
+  ]
+  # by default vsan_metric_skip_verify = false
+  vsan_metric_skip_verify = true
+  # vsan_metric_exclude = [ ]
```
Comment (Hipska) [inline plugins/inputs/vsphere/README.md:628]: I still think this needs to be uncommented, as the default is `*` which should exclude all the defined metrics?
```suggestion
  vsan_metric_exclude = [ ]
```

File: plugins/inputs/vsphere/README.md [hunk 16]
```diff
@@ -551,9 +561,231 @@ For a detailed list of commonly available metrics, please refer to
...
+  # Exclude all other metrics
+  vm_metric_exclude = ["*"]
+  datastore_metric_exclude = ["*"]
+  datacenter_metric_exclude = ["*"]
+  host_metric_exclude = ["*"]
+  cluster_metric_exclude = ["*"]
+  
+  vsan_metric_include = [ "performance.*" ]
+  vsan_metric_skip_verify = false
+
```
Comment (Hipska) [inline plugins/inputs/vsphere/README.md:781]: exclude is still `*` (default)?

File: plugins/inputs/vsphere/README.md [hunk 17]
```diff
@@ -551,9 +561,231 @@ For a detailed list of commonly available metrics, please refer to
...
+  # Exclude all other metrics
+  vm_metric_exclude = ["*"]
+  datastore_metric_exclude = ["*"]
+  datacenter_metric_exclude = ["*"]
+  host_metric_exclude = ["*"]
+  cluster_metric_exclude = ["*"]
+  
+  vsan_metric_include = [ "summary.*" ]
+  vsan_metric_skip_verify = false
+
```
Comment (Hipska) [inline plugins/inputs/vsphere/README.md:757]: exclude is still `*` (default)?

Comment (gangadharaswamy) [reply plugins/inputs/vsphere/README.md:757]: This may clarify your doubts - [performance-considerations](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/vsphere#performance-considerations)

File: plugins/inputs/vsphere/README.md (hunk 16 above)
Comment (gangadharaswamy) [reply plugins/inputs/vsphere/README.md:781]: This may clarify your doubts - [performance-considerations](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/vsphere#performance-considerations)

Comment (Hipska) [reply plugins/inputs/vsphere/README.md:781]: What do you suggest this will clarify? The mentioned metrics in that section aren't excluded by default. Here for vSAN the reverse logic needs to be applied.

File: plugins/inputs/vsphere/README.md (hunk 17 above)
Comment (Hipska) [reply plugins/inputs/vsphere/README.md:769]: Please update config so it is correct, just like on line 640.

File: plugins/inputs/vsphere/README.md (hunk 16 above)
Comment (Hipska) [reply plugins/inputs/vsphere/README.md:793]: Please update config so it is correct, just like on line 640.

**Review by Hipska — APPROVED** [review]
Thanks, the documentation now looks consistent.

File: plugins/inputs/vsphere/README.md [hunk 18]
```diff
@@ -771,7 +771,6 @@ For example:
   cluster_metric_exclude = ["*"]
   
   vsan_metric_include = [ "summary.*" ]
-  vsan_metric_exclude = [ ]
```
Comment (Hipska) [inline plugins/inputs/vsphere/README.md]: Why did you remove it again?

Comment (gangadharaswamy) [reply plugins/inputs/vsphere/README.md]: Added back the vsan_metric_exclude changes. Missed while checking DisconnectedServersBehavior build error.

File: plugins/inputs/vsphere/vsan.go [hunk 19]
```diff
@@ -0,0 +1,539 @@
...
+		}
+	}
+	return metrics
+}
+
+// getCmmdsMap returns a map which maps a uuid to a CmmdsEntity
+func getCmmdsMap(ctx context.Context, client *vim25.Client, clusterObj *object.ClusterComputeResource) (map[string]CmmdsEntity, error) {
+	hosts, err := clusterObj.Hosts(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("fail to get host: %v", err)
```
[the remark this answers is not in the document]
Comment (Hipska) [reply plugins/inputs/vsphere/vsan.go:159]: Linter is also complaining about this.

**Review by Hipska — APPROVED** [review]

File: plugins/inputs/vsphere/vsan.go [hunk 20]
```diff
@@ -0,0 +1,524 @@
...
+					// The return time string is in UTC time
+					utcTimeStamp := fmt.Sprintf("%sT%sZ", tsParts[0], tsParts[1])
+					ts, ok := time.Parse(time.RFC3339, utcTimeStamp)
+					if ok != nil {
+						e.Parent.Log.Errorf("[vSAN] Failed to parse a timestamp: %s. Skipping", utcTimeStamp)
+						timeStamps = append(timeStamps, time.Time{})
+						continue
+					}
+					timeStamps = append(timeStamps, ts)
+				}
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:276]: Why don't you specify your custom time format directly instead of splitting and combining stuff?

File: plugins/inputs/vsphere/vsan.go [hunk 21]
```diff
@@ -0,0 +1,524 @@
...
+			}
+			// 2. Iterate on each measurement
+			for _, counter := range em.Value {
+				metricLabel := internal.SnakeCase(counter.MetricId.Label)
+				// 3. Iterate on each data point.
+				for i, values := range strings.Split(counter.Values, ",") {
+					ts := timeStamps[i]
+					if ts == (time.Time{}) {
```
Comment (srebhan) [inline plugins/inputs/vsphere/vsan.go:284]: `time.IsZero()`?

**Review by srebhan — COMMENTED** [review]
@gangadharaswamy I do have some comments...

File: plugins/inputs/vsphere/vsan.go (hunk 21 above)
Comment (gangadharaswamy) [reply plugins/inputs/vsphere/vsan.go]: Updated with time.IsZero().

**Review by Hipska — APPROVED** [review]

**Comment by gangadharaswamy** [comment]
@srebhan Could you please review the PR?

**Comment by gangadharaswamy** [comment]
@srebhan What's the next step? Is there anything that needs to be done to move forward?

**Comment by powersj** [comment]
> 
@srebhan What's the next step? Is there anything that needs to be done to move forward?

Sven and I just went through all the outstanding PRs to see what we need to do. For this PR he needs to get you a review. nothing needed from your side.

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for your contribution @gangadharaswamy and sorry the delay!

**Review by powersj — APPROVED** [review]

**Comment by gangadharaswamy** [comment]
Hi @srebhan , @powersj,
To which Telegraf release this PR will be considered? Will it be part of alternative weekly releases or quarterly releases?

**Comment by Hipska** [comment]
There is a bugfix release every 3 weeks and a quarterly feature releas. Since this is not a bug fix , but a feature, it will be on the next feature release, which is planned at June 12th.
